### PR TITLE
Next gen middleware date

### PR DESCRIPTION
**Pull request by kamilafsar**
https://trello.com/c/qJMtosrt/26-client-should-marshall-date-strings-from-json-to-dates

### CODE REVIEW

File: packages/core/src/code-gen/parsers/generateReferenceParser.ts [hunk 1]
```diff
@@ -7,104 +7,95 @@ import {
...
+  return tsx.block(
+    tsx.const({
+      name: parseResultName,
+      init: generateParserCall(pointer),
+    }),
+    tsx.statement.if({
+      expression: tsx.expression.negate(
+        tsx.expression.propertyAccess(parseResultName, "ok"),
+      ),
+      // TODO populate the errors with the actual errors
```
Comment (Jpunt) [inline packages/core/src/code-gen/parsers/generateReferenceParser.ts:25]: Is this something we need to fix now, or write down to not forget?

File: packages/core/src/code-gen/parsers/generateTypeParameterParser.ts [hunk 2]
```diff
@@ -10,48 +10,53 @@ import {
...
+        { args: [pointer.dataVarExpr] },
+      ),
+    }),
 
-        ts.factory.createIdentifier("ok"),
+    tsx.statement.if({
+      expression: tsx.expression.negate(
+        tsx.expression.propertyAccess(tResultName, "ok"),
+      ),
+      // TODO populate the errors with the actual errors
```
Comment (Jpunt) [inline packages/core/src/code-gen/parsers/generateTypeParameterParser.ts:51]: +1

**Review by Jpunt — APPROVED** [review]
Wow, tsx reads like a charm 🙌 amazing work!

File: packages/core/src/code-gen/parsers/generateReferenceParser.ts (hunk 1 above)
Comment (kamilafsar) [reply packages/core/src/code-gen/parsers/generateReferenceParser.ts:25]: The TODO was already there. I intent to fix this later.

**Pull request merged**